src/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -39,5 +42,37 @@ public void SendAsync_SimpleGet_Success()
                 _output.WriteLine(responseContent);
             }
         }
+
+        [Fact]
+        [OuterLoop]
+        public void SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
+        {
+            var handler = new WinHttpHandler();
+            using (var client = new HttpClient(handler))
+            {
+                var cts = new CancellationTokenSource();
+                Task<HttpResponseMessage> t = client.GetAsync(SlowServer, cts.Token);
+
+                Thread.Sleep(500);
+                cts.Cancel();
+                
+                AggregateException ag = Assert.Throws<AggregateException>(() => t.Wait());
+                Assert.IsType<TaskCanceledException>(ag.InnerException);
+            }
+        }        
```

Comment:
Yes, I do plan to move them to HttpClient tests in the near future.  I was optimizing my time right now and only focusing on Windows issues in case the new tests fail on *Nix and I need to diagnose it.

The tests will need to be modified a little bit because of differences in the default timeout behaviors between HttpClientHandler and WinHttpHandler.  WinHttpHandler also supports more properties with more granular timeouts.